FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/workmanager/sync/DeleteDraftsWorker.kt

```diff
@@ -0,0 +1,118 @@
+/*
+ * © 2016-present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com
+ * Contributors: DenBond7
+ */
+
+package com.flowcrypt.email.jetpack.workmanager.sync
+
+import android.content.Context
+import androidx.work.Constraints
+import androidx.work.ExistingWorkPolicy
+import androidx.work.NetworkType
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.WorkerParameters
+import com.flowcrypt.email.BuildConfig
+import com.flowcrypt.email.api.email.FoldersManager
+import com.flowcrypt.email.api.email.gmail.GmailApiHelper
+import com.flowcrypt.email.api.email.model.LocalFolder
+import com.flowcrypt.email.database.MessageState
+import com.flowcrypt.email.database.entity.AccountEntity
+import com.flowcrypt.email.database.entity.MessageEntity
+import com.sun.mail.imap.IMAPFolder
+import jakarta.mail.Flags
+import jakarta.mail.Folder
+import jakarta.mail.Message
+import jakarta.mail.Store
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+/**
+ * @author Denis Bondarenko
+ *         Date: 9/30/22
+ *         Time: 10:51 AM
+ *         E-mail: DenBond7@gmail.com
+ */
+class DeleteDraftsWorker(context: Context, params: WorkerParameters) :
```

Comment:
How are the draft ids passed into this?

Reply:
Here is little different logic. We cache drafts locally and store `draftId` in the local database. Please look at the picture

![image](https://user-images.githubusercontent.com/2863246/194038913-55156d4d-e3b1-466f-88ec-e2e60796be93.png)